commit b2e81c0b025d9b259f055a63d025c929bd302935
Author: Cosimo Cecchi <cosimoc@gnome.org>
Date:   Sat Mar 21 11:30:53 2009 +0000

    fix the submenu example to make it work properly once installed. Patch by
    
    2009-03-21  Cosimo Cecchi  <cosimoc@gnome.org>
    
    	* examples/submenu.py: fix the submenu example to make it work
    	properly once installed.
    	Patch by Bruce van der Kooji (#559172).
    
    svn path=/trunk/; revision=61

diff --git a/ChangeLog b/ChangeLog
index 235d1a6..3467d2e 100644
--- a/ChangeLog
+++ b/ChangeLog
@@ -1,3 +1,9 @@
+2009-03-21  Cosimo Cecchi  <cosimoc@gnome.org>
+
+	* examples/submenu.py: fix the submenu example to make it work
+	properly once installed.
+	Patch by Bruce van der Kooji (#559172).
+
 === 0.5.1 ===
 2008-09-16  Johan Dahlin  <johan@gnome.org>
 
diff --git a/examples/submenu.py b/examples/submenu.py
index a2f2e30..9a98ebf 100644
--- a/examples/submenu.py
+++ b/examples/submenu.py
@@ -1,19 +1,23 @@
 import nautilus
 
 class ExampleMenuProvider(nautilus.MenuProvider):
-
+    
+    # Nautilus crashes if a plugin doesn't implement the __init__ method.
+    # See Bug #374958
+    def __init__(self):
+        pass
+        
     def get_file_items(self, window, files):
-        menuitem = nautilus.MenuItem('ExampleMenuProvider::Foo', 'Foo', '')
+        top_menuitem = nautilus.MenuItem('ExampleMenuProvider::Foo', 'Foo', '')
 
         submenu = nautilus.Menu()
-        menuitem.set_submenu(submenu)
+        top_menuitem.set_submenu(submenu)
 
-        menuitem = nautilus.MenuItem('ExampleMenuProvider::Bar', 'Bar', '')
-        submenu.append_item(menuitem)
+        sub_menuitem = nautilus.MenuItem('ExampleMenuProvider::Bar', 'Bar', '')
+        submenu.append_item(sub_menuitem)
 
-        return menuitem,
+        return top_menuitem,
 
-    # FIXME: Why isn't this working?
     def get_background_items(self, window, file):
         submenu = nautilus.Menu()
         submenu.append_item(nautilus.MenuItem('ExampleMenuProvider::Bar', 'Bar', ''))
